JS/see_article.js

```diff
@@ -0,0 +1,179 @@
+    box_select_plus.className = "plus";
+
+    let box_select_value = document.createElement('input');
+    box_select_value.type = "number";
+    box_select_value.id = "quantity";
+    box_select_value.value = 1;
+    box_selectdiv.appendChild(box_select_minus);
+    box_selectdiv.appendChild(box_select_value);
+    box_selectdiv.appendChild(box_select_plus);
+
+    /*
+     * When we click on the plus button, the quantity of the article is updated in the view.
+     */
+    box_select_plus.onclick = function () {
+        document.getElementById('quantity').value = parseInt(document.getElementById('quantity').value) + 1;
+    };
+    /*
+     * When we click on the minus button, the quantity of the article is updated in the view as long as the value is superior to 0.
+     */
+    box_select_minus.onclick = function () {
+        if (!(parseInt(document.getElementById('quantity').value) - 1 == 0)) {
+            document.getElementById('quantity').value = parseInt(document.getElementById('quantity').value) - 1;
+        }
+    };
+
+    add_btnContent = document.createTextNode('Ajouter au pannier');
+
+    container_product_div.appendChild(box_article);
+    imagediv.appendChild(image);
+    name.appendChild(nameContent);
+    brand.appendChild(brandContent);
+    price.appendChild(priceContent);
+    descriptionP.appendChild(descriptionPContent);
+    box_article.appendChild(imagediv);
+    box_article.appendChild(infoDiv);
+    infoDiv.appendChild(name);
+    infoDiv.appendChild(brand);
+    infoDiv.appendChild(detailsDiv);
+    detailsDiv.appendChild(price);
+    detailsDiv.appendChild(buttonDiv);
+    detailsDiv.appendChild(descriptionP);
+    add_btn.appendChild(add_btnContent);
+    buttonDiv.appendChild(box_selectdiv);
+    buttonDiv.appendChild(emptydiv);
+    buttonDiv.appendChild(add_btn);
+
+/*
+     This function add to the cart the article by taking the following informations:
+     - what is the article
+     - in what quantity is the item purchased
+     → Since our cart is saved in the sessionStorage, in order to have all the stuff put in there, we need to read the sessionStorage and then add the new element after verifying if it alreay exist in the cart or not
+
+    * old_data_saved : the var that contains all the informations stored in the sessionStorage
+    * quantity_article : the quantity of the item purchased
+    * tab : the array that contains all the json array to save into the sessionStorage.
+    * article_already_in_cart : has the value true if the article already exist in the cart of false in the other way around.
+    */
+
+    add_btn.onclick = function () {
+        var quantity_article = document.getElementById('quantity').value;
+        var old_data_saved = JSON.parse(sessionStorage.getItem('articleToCart2'));
+        var tab = [];
+        var article_already_in_cart = false;
+        if (old_data_saved != null) {
+            for (i = 0; i < Object.keys(old_data_saved).length; i++) {
```

Comment:
yet another weird loop :)

Reply:
I don't use "object.keys" anymore in this file.